### ISSUE TITLE
Question about backbone frame construction : Why the rotation matrix of literature amino acid backbone structure  is not identity matrix?

### ISSUE THREAD

**Issue by hw-protein** [aqlaboratory/openfold#145]
Hi there, I'm a student, studying openfold code these days.

 What makes me confused is that, the rotation matrix(or quaternion) of backbone structure of literature structure(defined by rigid_group_atom_positions) is not an identity matrix.

From openfold/openfold/np/residue_constants.py
```
rigid_group_atom_positions = {
    "ALA": [
        ["N", 0, (-0.525, 1.363, 0.000)],
        ["CA", 0, (0.000, 0.000, 0.000)],
        ["C", 0, (1.526, -0.000, -0.000)],
        ["CB", 0, (-0.529, -0.774, -1.205)],
        ["O", 3, (0.627, 1.062, 0.000)],
    ],
```
And from openfold/openfold/utils/rigid_utils.py
```
    @staticmethod
    def from_3_points(
        p_neg_x_axis: torch.Tensor, 
        origin: torch.Tensor, 
        p_xy_plane: torch.Tensor, 
        eps: float = 1e-8
    ) -> Rigid:
        """
            Implements algorithm 21. Constructs transformations from sets of 3 
            points using the Gram-Schmidt algorithm.

            Args:
                p_neg_x_axis: [*, 3] coordinates
                origin: [*, 3] coordinates used as frame origins
                p_xy_plane: [*, 3] coordinates
                eps: Small epsilon value
            Returns:
                A transformation object of shape [*]
        """
        p_neg_x_axis = torch.unbind(p_neg_x_axis, dim=-1)
        origin = torch.unbind(origin, dim=-1)
        p_xy_plane = torch.unbind(p_xy_plane, dim=-1)

        e0 = [c1 - c2 for c1, c2 in zip(origin, p_neg_x_axis)]
        e1 = [c1 - c2 for c1, c2 in zip(p_xy_plane, origin)]

        denom = torch.sqrt(sum((c * c for c in e0)) + eps)
        e0 = [c / denom for c in e0]
        dot = sum((c1 * c2 for c1, c2 in zip(e0, e1)))
        e1 = [c2 - c1 * dot for c1, c2 in zip(e0, e1)]
        denom = torch.sqrt(sum((c * c for c in e1)) + eps)
        e1 = [c / denom for c in e1]
        e2 = [
            e0[1] * e1[2] - e0[2] * e1[1],
            e0[2] * e1[0] - e0[0] * e1[2],
            e0[0] * e1[1] - e0[1] * e1[0],
        ]

        rots = torch.stack([c for tup in zip(e0, e1, e2) for c in tup], dim=-1)
        rots = rots.reshape(rots.shape[:-1] + (3, 3))

        rot_obj = Rotation(rot_mats=rots, quats=None)

        return Rigid(rot_obj, torch.stack(origin, dim=-1))
```
With the above coordinates of Alanine backbone and rigids_from_3_point, the rotation matrix of literature backbone structure  is ```[ [-1, 0 , 0] , [0, 1, 0],  [0, 0, -1]]```, which is not an identity rotation matrix.
 `p_neg_axis` is position of C, `origin` is position of CA, `p_xy_plane` is position of N, from the section 1.8.1 of the supplementary of the AlphaFold2 paper.
 Below is code to check it.
```
ala_lit_coords=torch.Tensor([ [-0.525, 1.363, 0.000],[0.0000,0.000,0.000],[1.526,0.000,0.000]]).float()
ref_frame=Rigid.from_3_points(
                            p_neg_x_axis=ala_lit_coords[..., 2, :],
                            origin=ala_lit_coords[...,1,:],
                            p_xy_plane=ala_lit_coords[...,0,:],
                            eps=1e-4
                            )
print (ref_frame._rots._rot_mats)
#tensor([[-9.9998e-01, -1.6530e-05,  0.0000e+00],
        [ 0.0000e+00,  9.9997e-01,  0.0000e+00],
        [ 0.0000e+00,  0.0000e+00, -9.9995e-01]])
```

 Since the rotation matrix of the literature backbone structure (reference rotation matrix) is not identity matrix, when we multiply reference matrix by new rotation matrix , the resulting global backbone frame rotation value is different from the rotation matrix we multiplied.

 In other words, even if the ground-true rotation matrix of the backbone frame is given, the rotation matrix of the resulting backbone is different from the given ground-true rotation matrix because the rotation matrix of the literature backbone structure is not an identity matrix.
i.e ( R_global_backbone = R_truth * R_literature , but , R_global_backbone != R_truth, because, R_literature != Identity matrix)

I think it is weird... 

 However, since the FAPE loss is computed with the atom position xj relative to the frame Ti, I don't think the issue I raised is a problem. 

 My English writing skill is not good enough, so I'm worried that what I'm trying to convey has been conveyed well. If possible, I'd like to know your thoughts on this issue.

**Comment by jamaliki** [comment]
If you take a look at the AlphaFold code, they mirror the x and z axis. I think it is in protein.py.
